update docs with X-Druid-SQL-Query-Id (#11761)
apache/druid: docs/querying/sql.md

{"query" : "SELECT COUNT(*) FROM data_source WHERE foo = 'bar'"}
```

##### Request body
      
|Property|Description|Default|
|--------|----|-----------|
|`query`|SQL query string.| none (required)|
|`resultFormat`|Format of query results. See [Responses](#responses) for details.|`"object"`|

> |`header`|Whether or not to include a header. See [Responses](#responses) for details.|`false`|

Reviewer: Maybe clarify further that this means header row for the result set, not having to do with the HTTP header.

> |`header`|Whether or not to include a header for the query result. See [Responses](#responses) for details.|`false`|

Reviewer: ```suggestion
|`header`|Whether or not to include a header row for the query result. See [Responses](#responses) for details.|`false`|
```
I think saying header row helps with the differentiation.

|--------|----|-----------|
|`query`|SQL query string.| none (required)|
|`resultFormat`|Format of query results. See [Responses](#responses) for details.|`"object"`|

> |`header`|Whether or not to include a header for the query response. See [Responses](#responses) for details.|`false`|

Reviewer 2: wait. I think there is some confusion here. The `header` here is talking about the column names of each column in the result. This is different for the header of the HTTP response.

Author: @maytasm Yeah, that's what I was trying to differentiate but not very well. Maybe "query result" would be less confusing?

}
```

> In this case, the first result of the response body returned will be a header. For the `csv`, `array`, and `arrayLines` formats, the header

Reviewer: ```suggestion
In this case, the first result of the response body is the header row. For the `csv`, `array`, and `arrayLines` formats, the header
```

In this case, the first result returned will be a header. For the `csv`, `array`, and `arrayLines` formats, the header
will be a list of column names. For the `object` and `objectLines` formats, the header will be an object where the
keys are column names, and the values are null.
If you specified `sqlQueryId` in the [connection context parameters](#connection-context),

> the SQL query identifier will be returned in the header labeled `X-Druid-SQL-Query-Id`.

Reviewer: ```suggestion
Druid returns the SQL query identifier in the `X-Druid-SQL-Query-Id` header.
```
nit: prefer active

Reviewer 2: If you don't specified `sqlQueryId` in the connection context parameters, a sqlQueryId will be generated by Druid and return in the response's `X-Druid-SQL-Query-Id` header
